doc/source/ray-core/gpu-objects.rst

```diff
@@ -0,0 +1,212 @@
+Adding this decorator will change Ray's behavior in the following ways:
+
+1. When returning the tensor, Ray will store a *reference* to the tensor instead of copying it to CPU memory. If multiple tensors are found, then Ray will store a reference for each tensor.
+2. When the :class:`ray.ObjectRef` is passed to another task, Ray will use Gloo to transfer the tensor to the destination task.
+
+For (2) to work, we must first create a *collective group* of actors.
+
+Creating a collective group
+***************************
+
+.. Add API pages for collective group APIs.
+
+To create a collective group for use with GPU objects, we must:
+
+1. Create multiple Ray actors.
+2. Create a collective group on the actors using the :func:`ray.experimental.collective.create_collective_group <ray.experimental.collective.create_collective_group>` function. The tensor transport specified should match the one used in the :func:`@ray.method <ray.method>` decorator.
+
+Here is an example:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_group_start__
+   :end-before: __gloo_group_end__
+
+The actors can now communicate directly via gloo.
+The group can also be destroyed using the :func:`ray.experimental.collective.destroy_collective_group <ray.experimental.collective.destroy_collective_group>` function.
+After calling this function, the actors can be reused as normal and new collective groups can be created on them.
+
+Passing GPU objects to other actors
+***********************************
+
+Now that we have a collective group, we can create and pass GPU objects between the actors.
+Here is a full example:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_full_example_start__
+   :end-before: __gloo_full_example_end__
+
+When the :class:`ray.ObjectRef` is passed to another task, Ray will use Gloo to transfer the tensor directly from the source actor to the destination actor instead of the default object store.
+This is done by submitting an additional Ray task to each actor, which executes the send and receive operations on a background thread.
+
+Passing GPU objects to the same actor
+*************************************
+
+GPU :class:`ray.ObjectRefs <ray.ObjectRef>` can also be passed to the same actor.
+This avoids any copies and just returns the ``torch.Tensors`` directly back to the same actor.
+For example:
+
+.. literalinclude:: doc_code/gpu_objects_gloo.py
+   :language: python
+   :start-after: __gloo_intra_actor_start__
+   :end-before: __gloo_intra_actor_end__
+
+
+.. note::
+    Ray only keeps a reference to the tensor created by the user, so the tensor objects are *mutable*.
+    If ``sender.sum`` were to modify the tensor in the above example, the changes would also be seen by ``receiver.sum``.
+    This differs from the normal Ray Core API, which always makes an immutable copy of data returned by actors.
```

Comment:
I feel like this needs to be highlighted more strongly somewhere. Maybe at the top of the page we should have a "TL;DR" that outlines the key points:

- You need to create a collective group and specify a transport
- Objects are mutable until they're consumed
- Otherwise actors can be used as normal

Reply:
I see, it's outlined in more detail in the limitations section. that is probably sufficient.

still might be nice to have a TL;DR section (and it could link to limitations)